unittests/backends/QPPTester.cpp

```diff
@@ -1241,3 +1280,75 @@ CUDAQ_TEST(QPPTester, checkReset) {
     EXPECT_EQ(1, qppBackend.mz(q1));
   }
 }
+
+CUDAQ_TEST(QPPTester, checkScrap) {
+
+  // State Vector. Just to show the desired behavior.
+  {
+    // Initialize QPP Backend with 1 qubit.
+    QppCircuitSimulator<qpp::ket> qppBackend;
+    auto q0 = qppBackend.allocateQubit();
+    EXPECT_EQ(0, qppBackend.mz(q0));
+    
+    // Rotate to |1>
+    qppBackend.x(q0);
+    EXPECT_EQ(1, qppBackend.mz(q0));
+
+    // Add another qubit to the system. Individually, should be |0>.
+    auto q1 = qppBackend.allocateQubit();
+    EXPECT_EQ(0, qppBackend.mz(q1));
+
+    std::string got_bitstring = getSampledBitString(qppBackend, {0,1});
+    EXPECT_EQ("10", got_bitstring);
+    EXPECT_EQ(1, qppBackend.mz(q0));
+    EXPECT_EQ(0, qppBackend.mz(q1));
+  }
+
+  // Density Matrix. Allocate qubits at same time.
+  // The qubit ordering aligns with the state vector backend
+  // and all preexisting tests in this file.
+  {
+    // Initialize QPP Backend with 2 qubits together.
+    QppNoiseCircuitSimulator qppBackend;
+    auto q0 = qppBackend.allocateQubit();
+    auto q1 = qppBackend.allocateQubit();
+    EXPECT_EQ(0, qppBackend.mz(q0));
+    EXPECT_EQ(0, qppBackend.mz(q1));
+
+    // Rotate q0 to |1>
+    qppBackend.x(q0);
+    EXPECT_EQ(1, qppBackend.mz(q0));
+
+    std::string got_bitstring = getSampledBitString(qppBackend, {0,1});
+    EXPECT_EQ("10", got_bitstring);
+    EXPECT_EQ(1, qppBackend.mz(q0));
+    EXPECT_EQ(0, qppBackend.mz(q1));
+  }
+
+  // Density Matrix. Add new qubit later.
+  // The qubit ordering gets flipped in this case.
+  {
+    // Initialize QPP Backend 1 qubit at a time.
+    QppNoiseCircuitSimulator qppBackend;
+    auto q0 = qppBackend.allocateQubit();
+    // Passes.
+    EXPECT_EQ(0, qppBackend.mz(q0));
+    
+    // Rotate to |1>
+    qppBackend.x(q0);
+    // Passes.
+    EXPECT_EQ(1, qppBackend.mz(q0));
+
+    // Add another qubit. Individually, should be |0>.
+    auto q1 = qppBackend.allocateQubit();
+    // Fails. Qubits have flipped. Even when directly addressing
+    // the variable for q1.
```

Comment:
It actually seems that the variable has entirely flipped as well. If you do the following instead of measure
```
    qppBackend.x(q1);
```
the 2 tests above suggest we should now be in the |11> state. Instead, we return to |00>, as if `q1` is now addressing the qubit that was previously stored in `q0`.